sdl_android/src/main/java/com/smartdevicelink/protocol/SdlProtocol.java

```diff
@@ -0,0 +1,1319 @@
+            }else if(serviceType.equals(SessionType.NAV)){
+                mtuTag = ControlFrameTags.Video.StartServiceACK.MTU;
+            }
+            Object mtu = packet.getTag(mtuTag);
+            if(mtu!=null){
+                mtus.put(serviceType,(Long) packet.getTag(mtuTag));
+            }
+            if(serviceType.equals(SessionType.RPC)){
+                hashID = (Integer) packet.getTag(ControlFrameTags.RPC.StartServiceACK.HASH_ID);
+                Object version = packet.getTag(ControlFrameTags.RPC.StartServiceACK.PROTOCOL_VERSION);
+
+                if(version!=null) {
+                    //At this point we have confirmed the negotiated version between the module and the proxy
+                    protocolVersion = new Version((String) version);
+                }else{
+                    protocolVersion = new Version("5.0.0");
+                }
+
+                //Check to make sure this is a transport we are willing to accept
+                TransportRecord transportRecord = packet.getTransportRecord();
+
+                if(transportRecord == null || !requestedPrimaryTransports.contains(transportRecord.getType())){
+                    onTransportNotAccepted("Transport is not in requested primary transports");
+                    return;
+                }
+
+                boolean activeTransportsHandled = false;
+
+                // This enables custom behavior based on protocol version specifics
+                if (protocolVersion.isNewerThan(new Version("5.1.0")) >= 0) {
+
+                    if (activeTransports.get(SessionType.RPC) == null) {    //Might be a better way to handle this
+
+                        ArrayList<String> secondary = (ArrayList<String>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.SECONDARY_TRANSPORTS);
+                        ArrayList<Integer> audio = (ArrayList<Integer>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.AUDIO_SERVICE_TRANSPORTS);
+                        ArrayList<Integer> video = (ArrayList<Integer>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.VIDEO_SERVICE_TRANSPORTS);
+
+                        activeTransports.put(SessionType.RPC, transportRecord);
+                        activeTransports.put(SessionType.BULK_DATA, transportRecord);
+                        activeTransports.put(SessionType.CONTROL, transportRecord);
+
+                        //Build out the supported secondary transports received from the
+                        // RPC start service ACK.
+                        supportedSecondaryTransports = new ArrayList<>();
+                        if (secondary == null) {
+                            // If no secondary transports were attached we should assume
+                            // the Video and Audio services can be used on primary
+                            if (requiresHighBandwidth
+                                    && TransportType.BLUETOOTH.equals(transportRecord.getType())) {
```

Comment:
Should be checking the `audio` and `video` lists in this case, rather than the transport config.

Reply:
Per a side discussion, this is the desired logic. If there are no secondary transports available but the developer specified they required high bandwidth, then the the session should be terminated. They will be notified if a new transport connects.